src/DrupalFinder.php

```diff
@@ -28,53 +30,26 @@ public function locateRoot($start_path) {
     $this->composerRoot = FALSE;
 
     foreach (array(TRUE, FALSE) as $follow_symlinks) {
-      $path = $start_path;
-      if ($follow_symlinks && is_link($path)) {
-        $path = realpath($path);
-      }
-      // Check the start path.
-      if ($checked_path = $this->isValidRoot($path)) {
-        return $checked_path;
-      }
-      else {
-        // Move up dir by dir and check each.
-        while ($path = $this->shiftPathUp($path)) {
-          if ($follow_symlinks && is_link($path)) {
-            $path = realpath($path);
-          }
-          if ($checked_path = $this->isValidRoot($path)) {
-            return $checked_path;
-          }
+      $path = new SplFileInfo($start_path);
+      do {
+        if ($follow_symlinks && $path->isLink()) {
+          $path = new SplFileInfo($path->getRealPath());
+        }
+        // Check the start path.
+        if ($this->isValidRoot($path->getPathname())) {
+          return TRUE;
         }
       }
+      while (($path = $path->getPathInfo()) && ($path->getFilename() != '.'));
```

Comment:
This works only with a streamwrapper... not a real FS... it get's into a loop at fs-root.